### PR TITLE
Incorporated suggested changes from #1600

### PR DESCRIPTION
**Pull request by PeterKraus**
## Description
Addresses issues highlighted in #1600:
- energies and radii are now sorted based on radius of the diatomic
- initial guess for `r_e` is now the point of minimum energy
- check for number of points around minimum: if fewer than three on either side, print out a warning.

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [x] Addresses #1600 

## Questions
- [ ] Handling warnings? Do we need a special tag?

## Checklist
- [ ] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [ ] Ready for merge

### CODE REVIEW

File: psi4/driver/diatomic.py [hunk 1]
```diff
@@ -149,12 +162,13 @@ def anharmonicity(rvals, energies, plot_fit='', mol = None):
     B0 = B - ae / 2.0
     r0 = np.sqrt(constants.h / (8.0 * np.pi**2 * mu * constants.c * B0))
     recheck = np.sqrt(constants.h / (8.0 * np.pi**2 * mu * constants.c * B))
-    r0 /= angstrom_to_meter;
-    recheck /= angstrom_to_meter;
+    r0 /= angstrom_to_meter
+    recheck /= angstrom_to_meter
```
Comment (andysim) [inline psi4/driver/diatomic.py]: Hah!  You can tell I have a hard time switching between C++ and Python sometimes :)

File: psi4/driver/diatomic.py [hunk 2]
```diff
@@ -149,12 +162,13 @@ def anharmonicity(rvals, energies, plot_fit='', mol = None):
...
 
     # Fundamental frequency nu
-    nu = we - 2.0 * wexe;
-    zpve_nu = 0.5 * we - 0.25 * wexe;
+    nu = we - 2.0 * wexe
+    zpve_nu = 0.5 * we - 0.25 * wexe
+    zpve_we = 0.5 * we
```
Comment (andysim) [inline psi4/driver/diatomic.py]: This is a good idea; it allows the user to see the effect of anharmonicity

File: psi4/driver/diatomic.py [hunk 3]
```diff
@@ -249,7 +265,8 @@ def anharmonicity(rvals, energies, plot_fit='', mol = None):
                "we"               :  we,
                "wexe"             :  wexe,
                "nu"               :  nu,
-               "ZPVE(harmonic)"   :  zpve_nu,
+               "E"                :  e,
```
Comment (andysim) [inline psi4/driver/diatomic.py]:
```suggestion
               "E at re"           :  e,
```

File: psi4/driver/diatomic.py [hunk 4]
```diff
@@ -104,9 +109,17 @@ def anharmonicity(rvals, energies, plot_fit='', mol = None):
     m1 = molecule.mass(0)
     m2 = molecule.mass(1)
 
+    # Find rval of the minimum of energies, check number of points left and right
+    minE = np.min(energies)
+    iminE = energies.index(minE)
```
Comment (dgasmith) [inline psi4/driver/diatomic.py]: Can we switch this to `minE = energies.argmin()`? Worried a bit about comparing floats.

File: psi4/driver/diatomic.py [hunk 5]
```diff
@@ -104,9 +109,17 @@ def anharmonicity(rvals, energies, plot_fit='', mol = None):
     m1 = molecule.mass(0)
     m2 = molecule.mass(1)
 
+    # Find rval of the minimum of energies, check number of points left and right
+    minE = np.min(energies)
```
Comment (dgasmith) [inline psi4/driver/diatomic.py]: Can this be `minimum_energy`? Too many short hands as it is, can get to compressed. Ditto for `iminE`

File: psi4/driver/diatomic.py [hunk 6]
```diff
@@ -95,6 +95,11 @@ def anharmonicity(rvals, energies, plot_fit='', mol = None):
         raise ValidationError("At least 5 data points must be provided to compute anharmonicity")
     core.print_out("\n\nPerforming a fit to %d data points\n" % npoints)
 
+    # Sort radii and values first from lowest to highest radius
+    sortedtuples = sorted(zip(rvals,energies))
```
Comment (dgasmith) [inline psi4/driver/diatomic.py]:
```
idx = np.argsort(rvals)
rvals = rvals[idx]
energies = energies[idx]
```

File: psi4/driver/diatomic.py [hunk 7]
```diff
@@ -95,6 +95,11 @@ def anharmonicity(rvals, energies, plot_fit='', mol = None):
...
+    rvals = [i[0] for i in sortedtuples]
+    energies = [i[1] for i in sortedtuples]
+
     # Make sure the molecule the user provided is the active one
     molecule = mol if mol is not None else core.get_active_molecule()
```
Comment (dgasmith) [inline psi4/driver/diatomic.py]:
```python
molecule = mol or core.get_active_molecule()
```

File: psi4/driver/diatomic.py (hunk 6 above)
Comment (PeterKraus) [reply psi4/driver/diatomic.py]: I had to wrap the python arrays around `np.array()`, hope that's ok.

**Pull request merged**